tests/typing/api.py

```diff
@@ -323,6 +323,19 @@ async def typecheck_stdlib_async() -> None:
     await logger.alog(logging.CRITICAL, "async log")
 
 
+def typecheck_bound_logger_return() -> None:
+    blogger: structlog.BoundLogger = structlog.get_logger(__name__)
```

Comment:
Interesting find here, and perhaps the reason that this hasn't come up sooner: 
This test would fail with `structlog.BoundLogger` before this PR, but it would pass with `structlog.stdlib.BoundLogger`.

Reply:
Mmm, after hitting this, I think this is actually the root of the issue I'm experiencing; I don't think I realized that in some places I was working with a `structlog.BoundLogger` and other places I was working with a `structlog.stdlib.BoundLogger`, and if I did, I did not expect there to be a difference. My assumption would have been that the former was a top-level export of the latter. At least then I think my type error woes are gone. I'll still get this PR over the finish line, since i think it's correct and valuable.